DASH-IF-IOP.bs.md

```diff
@@ -10,7 +10,242 @@ Live discussion in #document-authoring on Slack.
+
+Note: It may be impractical to present [=media segments=] only partially, depending on the capabilties of the client platform, the type of media samples involved and any dependencies between samples. It is up to the client to ensure that platform capabilities are not exceeded and to account for the time shift that it incurs due to over-/underplayback.
+
+## Sample timeline ## {#timing-sampletimeline}
+
+The samples within a [=representation=] exist on a <dfn>sample timeline</dfn> defined by the encoder that created the samples. One or more sample timelines are mapped onto the [=MPD timeline=] by metadata stored in or referenced by the [=MPD=].
+
+The sample timeline SHALL be shared by all [=representations=] in the same adaptation set. [=Representations=] in different adaptation sets MAY use different sample timelines.
+
+The sample timeline is measured in unnamed timescale units. The term timescale refers to the number of timescale units per second. This value may be present in the [=MPD=] as the `@timescale` attribute, [[#timing-addressingmodes|depending on the addressing mode used]].
+
+<figure>
+	<img src="Images/Timing/PresentationTimeOffset.png" />
+	<figcaption>`@presentationTimeOffset` establishes the relationship between the [=MPD timeline=] and the sample timeline.</figcaption>
+</figure>
+
+The point on the sample timeline indicated by `@presentationTimeOffset` (in timescale units, default zero) SHALL be considered equivalent to the [=period=] start point on the [=MPD timeline=].
+
+If a `SegmentTemplate` [[#timing-addressingmodes|addressing mode]] is used, `@presentationTimeOffset` SHALL be a point within or at the start of the first [=media segment=] referenced by the [=period=] (even if the first [=media segment=] is no longer available or no longer referenced by the [=MPD=], as may be the case with a dynamic MPD).
+
+If a `SegmentBase` [[#timing-addressingmodes|addressing mode]] is used, `@presentationTimeOffset` SHALL be a point within or at the start of any [=media segment=] referenced by the [=period=].
+
+## Media segments ## {#timing-mediasegment}
+
+A <dfn>media segment</dfn> is an HTTP-addressable data structure that contains one or more media samples.
+
+Note: Parts of [[MPEGDASH]] use the term "subsegment" instead of "segment". The distinction is largely irrelevant in practice.
+
+[=Media segments=] SHALL contain one or more consecutive media samples.
+
+[=Media segments=] SHALL contain the media samples that exactly match the time span on the [=sample timeline=] that is mapped to the segment's associated time span on the [=MPD timeline=], except when the [[#timing-addressingmodes|addressing mode]] in use allows for inaccuracy in [=media segment=] contents.
+
+### Inaccurate addressing ### {#timing-mediasegment-inaccuracy}
```

Comment:
I believe this type of addressing is fairly common practice and is also implicitly (probably even explicitly) endorsed by IOP v4.2, which is why I put this chapter into the text.

I have seen a fair bit of content with the keyframe distance and the segment size not aligned up, while at the same time using SegmentTemplate without SegmentTimeline addressing that requires a fixed segment duration. This ends up with the sort of drift described here, where segments are only timed on average. In practice, players appear to handle this fine in my experience, although I have never tested seeking accuracy as such in these conditions (and I would not expect accurate seeking).

That being said, I am not opposed to taking a stricter view on this if it causes excessive client implementation complexity. Certainly, packagers should bear the responsibility of making easy to process content. Let's discuss this in detail in today's call.

Reply:
Ack - Let's discuss today.

ExoPlayer currently assumes that the manifest is accurate (we don't have any logic for the inefficient-but-accurate seek as I describe above), and we've never had any complaints.

If I had to guess, my assumption would be that this is because content is a lot more accurate in practice than this section allows for (the 50% deviation specified below seems very large). However it's possible that I've just been living in ignorant bliss, that content is often quite inaccurate, and that the player happens to do a good enough job to avoid complaints!

Reply:
We did not get to this part in the call yesterday due to running out of time. However, the slides by Thomas also outlined the inaccurate scenario, so I suspect it is taken as a natural use case by many. The fact that ExoPlayer expects accuracy is an important signal that perhaps there is grounds to revisit the topic, though.

If you feel that this topic deserves further exploration (either as a whole or even just the 50% value), I suggest opening a separate IOP issue for this and opening separate discussion on that topic in the IOP mailing list and/or the next IOP call. I will be happy to integrate any results from that discussion into this PR, of course.